### DISCUSSION TITLE
Transmission Models

### DISCUSSION THREAD

**Discussion by DemingCheng** [eclipse/mosaic#118] [Q&A] [answered]
Hi all.
I noticed that there are Transmission Models( greedy-forwarding and flooding algorithm ) in SNS,does the cell has the similar models?
BTW, is it possiable to use the ns-3 to simulate the routing algorithm, such as OLSR, with eclipse mosaic?

I'm a rookie in both mosaic and ns-3, so I’m eager for suggestions from everyone. 
Thank you!!

**Comment by schwepmo** [comment, marked as answer]
To answer your first question: The SNS is only used for AdHoc communication (direct communication from unit to unit), while the cell tries to abstract a whole internet infrastructure, complete with IP-based communication. Therefore the Cell separates the process of message transmission into three abstraction modules: `UpstreamModule`, `GeocasterModule` & `DownstreamModule` (see [here](https://www.eclipse.org/mosaic/docs/simulators/network_simulator_cell/) for further information). This level of abstraction is used to get good estimates of the delays, transmission issues, etc. that messages receive during transmission. However there is no focus on specific routing algorithms. So short answer: no, the cell doesn't use different routing algorithms.

Regarding your second question: Yes, in theory you should be able to use ns-3 however you like in combination with MOSAIC. That said, we're currently in the process of refactoring the installation of ns-3 and people are experiencing some issues with it. I'm personally not really experienced with using ns-3 so I can't give you more detailed insights.

**Reply by DemingCheng** [reply]
Thank you so much  for your help!
